core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,224 @@
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowLogs
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowZims
+import kotlinx.android.synthetic.main.activity_kiwix_error.reportButton
+import kotlinx.android.synthetic.main.activity_kiwix_error.restartButton
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.dao.NewBookDao
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.getCurrentLocale
+import org.kiwix.kiwixmobile.core.utils.files.FileLogger
+import org.kiwix.kiwixmobile.zim_manager.MountPointProducer
+import java.io.PrintWriter
+import java.io.StringWriter
+import javax.inject.Inject
+import kotlin.system.exitProcess
+
+private const val STATUS = 10
+private const val ZERO = 0
+
+open class ErrorActivity : BaseActivity() {
+  @Inject
+  lateinit var bookDao: NewBookDao
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @Inject
+  lateinit var mountPointProducer: MountPointProducer
+
+  @Inject
+  lateinit var fileLogger: FileLogger
+
+  private lateinit var body: String
+  private var exception: Throwable? = null
+  private lateinit var emailIntent: Intent
+  private val deviceDetails = """Device Details:
+        Device:[${Build.DEVICE}]
+        Model:[${Build.MODEL}]
+        Manufacturer:[${Build.MANUFACTURER}]
+        Time:[${Build.TIME}]
+        Android Version:[${Build.VERSION.RELEASE}]
+        App Version:[$versionName $versionCode]
+        
+        """
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_kiwix_error)
+    val callingIntent = intent
+    val extras = callingIntent.extras
+    exception = if (extras != null && safeContains(extras)) {
+      extras.getSerializable(EXCEPTION_KEY) as Throwable
+    } else {
+      null
+    }
+    reportButton.setOnClickListener {
+      emailIntent = Intent(Intent.ACTION_SEND)
+      with(emailIntent) {
+        type = "vnd.android.cursor.dir/email"
+        putExtra(
+          Intent.EXTRA_EMAIL,
+          arrayOf("android-crash-feedback@kiwix.org")
+        )
+        putExtra(Intent.EXTRA_SUBJECT, subject)
+      }
+      body = initialBody
+      buildBody()
+      startActivityForResult(
+        Intent.createChooser(Intent().apply {
+          putExtra(Intent.EXTRA_TEXT, body)
+        }, "Send email..."), 1
+      )
+    }
+    restartButton.setOnClickListener { restartApp() }
+  }
+
+  private fun buildBody() {
+    if (allowLogs.isChecked) {
+      val file = fileLogger.writeLogFile(this)
+      val path =
+        FileProvider.getUriForFile(this, applicationContext.packageName + ".fileprovider", file)
+      with(emailIntent) {
+        addFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION)
```

Comment:
Adding intent extras is not part of `buildBody`

Reply:
so this should go in onCreate?

Reply:
maybe move to the above email intent extra-adding and make all that a method?

Reply:
okay let me try this

Reply:
> Adding intent extras is not part of `buildBody`

Please take a look at line number 114, I need path which is present in buildBody()

Reply:
@Frans-Lukas Can I shift first if statement in that method for emailIntent extra adding?

Reply:
yes! that should work.